src/test/java/org/elasticsearch/script/mustache/MustacheScriptEngineTest.java

```diff
@@ -98,14 +103,60 @@ public void testEscapeJson() throws IOException {
                 escaped.append(c);
             }
             StringWriter target = new StringWriter();
-            assertThat(escaped.toString(), equalTo(JsonEscapingMustacheFactory.escape(writer.toString(), target).toString()));
+            assertThat(escaped.toString(), equalTo(InternalMustacheFactory.escape(writer.toString(), target).toString()));
         }
     }
 
+    @Test @Repeat(iterations = 5)
```

Comment:
Instead of running the test for five iterations randomly why not deterministically iterate over the three options you want to test below?